### PR TITLE
🔧  add snapshotProposal

### PR DESCRIPTION
**Pull request by mlgarchery**
### Description of change

Adding the snapshotProposal to the marketEmission if it exists.

### Pull-Request Checklist

- [x] Code is up-to-date with the `main` branch
- [x] `yarn lint` passes with this change
- [x] `yarn test` passes with this change
- [ ] This pull request links relevant issues as `Fixes #0000` N/A
- [x] There are new or updated unit tests validating the change
- [ ] Documentation has been updated to reflect this change
- [x] The new commits follow conventions outlined in the [conventional commit spec](https://www.conventionalcommits.org/en/v1.0.0/)

### CODE REVIEW

File: test/marketsDistribution.test.ts [hunk 1]
```diff
@@ -11,4 +11,8 @@ describe("Markets distribution", () => {
     const marketDistribution = await getMarketsDistribution(now(), provider);
     expect(marketDistribution).not.toBeUndefined();
   });
+  it("Should now include the snapshotProposal field (true for every epoch starting at age3)", async () => {
+    const marketDistribution = await getMarketsDistribution();
```
Comment (julien-devatom) [inline test/marketsDistribution.test.ts:15]: Use a static timestamp to not make the test failing in the future. We cannot be sure today that the DAO will always use Snapshot votes mechanism for future ages.

**Review by julien-devatom — CHANGES_REQUESTED** [review]

File: test/marketsDistribution.test.ts (hunk 1 above)
Comment (mlgarchery) [reply test/marketsDistribution.test.ts:15]: if we deprecate snapshot we shouln't keep a test checking we use snapshot.

Comment (mlgarchery) [reply test/marketsDistribution.test.ts:15]: I would rather delete the test

Comment (julien-devatom) [reply test/marketsDistribution.test.ts:15]: but we want to keep the things working on the past epochs
So IMO it is a good thing to keep the test for the epoch with a snapshot proposal available

**Review by julien-devatom — APPROVED** [review]
LGTM

**Pull request merged**